pkg/db/db_test.go

```diff
@@ -0,0 +1,141 @@
+/*
+ * Nuts registry
+ * Copyright (C) 2019. Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package db
+
+import (
+	"encoding/json"
+	"github.com/lestrrat-go/jwx/jwa"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestOrganization_KeysAsSet(t *testing.T) {
+	valid := "{\"name\": \"Zorggroep Nuts 2\",\"identifier\": \"urn:oid:2.16.840.1.113883.2.4.6.1:00000001\",\"keys\": [{\"kty\":\"EC\",\"crv\":\"P-256\",\"x\":\"MKBCTNIcKUSDii11ySs3526iDZ8AiTo7Tu6KPAqv7D4\",\"y\":\"4Etl6SRW2YiLUrN5vfvVHuhp7x8PxltmWWlbbM4IFyM\",\"use\":\"enc\",\"kid\":\"2\"}]}"
+	invalidKeys := "{\"name\": \"Zorggroep Nuts 2\",\"identifier\": \"urn:oid:2.16.840.1.113883.2.4.6.1:00000001\",\"keys\": {\"kty\":\"EC\"}}"
+	
+	t.Run("No keys returns empty set", func(t *testing.T) {
+		o := Organization{}
+		set, err := o.KeysAsSet()
+		if assert.NoError(t, err) && assert.NotNil(t, set) {
+			assert.Len(t, set.Keys, 0)
+		}
+	})
+
+	t.Run("JWK set is parsed", func(t *testing.T) {
+		o := Organization{}
+		if assert.NoError(t, json.Unmarshal([]byte(valid), &o)) {
```

Comment:
The custom unmarshaler is being (at least) partially tested here, but I like to write a specific test for each custom marshaler/unmarshaler. It makes it more clear that all interesting cases have been considered, and can provide confidence when refactoring a (un)marshaler, for example.

Reply:
What is you opinion about testing generated code?

Reply:
I usually don't bother. So that does change my suggestion entirely.